jetty-documentation/src/main/asciidoc/contribution-guide/3-source.adoc

```diff
@@ -163,6 +163,60 @@ Intellectual Property is a hallmark concern of the Eclipse Foundation so you are
 As much as we would like to accept a tremendous pull request, without the proper chain of events being completed our hands are tied.
 That being said, the steps are not particularly onerous and we are happy to work with you to get them accomplished.
 
+==== Logging Conventions
+
+When deciding when and what to log, bear in mind a few things:
+
+* never use `LOG.debug` without a preceding `if (LOG.isDebugEnabled())`
+* we don't want to pollute the log with very long stacktraces unless necessary
+* we don't want to routinely produce logging events in response to data sent by a user
+* we should not call more than one LOG method for a single event: otherwise log messages may be interleaved and more confusing
```

Comment:
```suggestion
* we should not call more than one LOG method for a single event: otherwise log messages may be interleaved and more confusing
* we should never LOG.warn and then throw that exception, as that will result in double handling.
* we should seldom LOG.debug and then throw as that will make debug verbose and add little information.
```

Reply:
Added